File: 4.Clients/WebApi/ClientApp/src/entities/stage.ts
import { StageItem } from "./stageItem";
import { Consultant } from "./consultant";
import { Process } from "./process";
import { StageStatusEnum } from './enums/stage-status.enum';

export class Stage {
    id: number;
    date: Date;
    feedback: string;
    status: StageStatusEnum;
    consultantOwnerId: number;
    consultantDelegateId: number;
    processId: number;
    rejectionReason?: string;
}

File: 1.Domain/Domain.Model/Stage.cs
using Core;
using Domain.Model.Enum;
using System;
using System.Collections.Generic;

namespace Domain.Model
{
    public abstract class Stage : Entity<int>
    {
        public StageType Type { get; set; }

        public DateTime? Date { get; set; }

        public string Feedback { get; set; }

        public StageStatus Status { get; set; }

        public int ProcessId { get; set; }
        public Process Process { get; set; }
        public int? ConsultantOwnerId { get; set; }
        public Consultant ConsultantOwner { get; set; }
        public int? ConsultantDelegateId { get; set; }
        public Consultant ConsultantDelegate { get; set; }
        public string RejectionReason { get; set; }
    }
}